### DISCUSSION TITLE
Separate 'crossfade' backup agent

### DISCUSSION THREAD

**Discussion by voipster** [savonet/liquidsoap#3812] [General]
Greets all

On occasion I find myself making changes which necessitate restarting LS. So I find myself thinking about how to do this without terminating connections, especially if my main audio source is piped into LS via Harbor. So I think about setting up a simple relay with mksafe so that a quick restart won't drop connections.

And, if I'm going to do that, what would be really cool is if that simple in-out relay could detect the drop of source stream, and then (while there is still audio in the buffer), cross fade to a predetermined backup source. Then, when the audio re-appears in Harbor, crossfade back again to it.   

Is this achievable? And if so has anyone done it / got any suggestions on what that simple relay script might look like?

**Comment by toots** [comment]
Hi there! Yes this could be achieved. The important point to keep in mind is to make sure that `input.harbor` has enough buffer available to handle the transition. This should be the default now with a buffer of `12s` by default, making it possible to handle both fade in and fade out.
